From ea2e971efac93f4644ceede7b997081ed49780ad Mon Sep 17 00:00:00 2001
From: Fei Luo <axl.luo@gmail.com>
Date: Tue, 10 Feb 2026 14:00:00 +1300
Subject: [PATCH] Enable USART6 on STM32MP257F-DK for A35

Assign USART6 to CID1 (Cortex-A35/Linux) in RIFSC configuration and
set status to "okay" in the Linux device tree overlay.

---
 stm32mp2/a35-td/linux/stm32mp257f-dk-ca35tdcid-ostl.dts   | 2 +-
 stm32mp2/a35-td/optee/stm32mp257f-dk-ca35tdcid-ostl-rif.dtsi | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/stm32mp2/a35-td/linux/stm32mp257f-dk-ca35tdcid-ostl.dts b/stm32mp2/a35-td/linux/stm32mp257f-dk-ca35tdcid-ostl.dts
index 8c49050..39ad289 100644
--- a/stm32mp2/a35-td/linux/stm32mp257f-dk-ca35tdcid-ostl.dts
+++ b/stm32mp2/a35-td/linux/stm32mp257f-dk-ca35tdcid-ostl.dts
@@ -769,7 +769,7 @@
 	pinctrl-1 = <&usart6_idle_pins_a>;
 	pinctrl-2 = <&usart6_sleep_pins_a>;
 	uart-has-rtscts;
-	status = "disabled";
+	status = "okay";
 };
 
 &usb2_phy1 {
diff --git a/stm32mp2/a35-td/optee/stm32mp257f-dk-ca35tdcid-ostl-rif.dtsi b/stm32mp2/a35-td/optee/stm32mp257f-dk-ca35tdcid-ostl-rif.dtsi
index 601fc98..1e1f9c2 100644
--- a/stm32mp2/a35-td/optee/stm32mp257f-dk-ca35tdcid-ostl-rif.dtsi
+++ b/stm32mp2/a35-td/optee/stm32mp257f-dk-ca35tdcid-ostl-rif.dtsi
@@ -48,7 +48,7 @@
 		RIFPROT(STM32MP25_RIFSC_USART3_ID, EMPTY_SEMWL, RIF_LOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_EN, RIF_CFEN)
 		RIFPROT(STM32MP25_RIFSC_UART4_ID, EMPTY_SEMWL, RIF_LOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_EN, RIF_CFEN)
 		RIFPROT(STM32MP25_RIFSC_UART5_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_CID2, RIF_SEM_DIS, RIF_CFEN)
-		RIFPROT(STM32MP25_RIFSC_USART6_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_CID2, RIF_SEM_DIS, RIF_CFEN)
+		RIFPROT(STM32MP25_RIFSC_USART6_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_CID1, RIF_SEM_DIS, RIF_CFEN)
 		RIFPROT(STM32MP25_RIFSC_UART7_ID, EMPTY_SEMWL, RIF_LOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_EN, RIF_CFEN)
 		RIFPROT(STM32MP25_RIFSC_UART8_ID, EMPTY_SEMWL, RIF_LOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_EN, RIF_CFEN)
 		RIFPROT(STM32MP25_RIFSC_UART9_ID, EMPTY_SEMWL, RIF_LOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_EN, RIF_CFEN)
-- 
2.34.1

